src/main/java/duke/parser/InputParser.java

```diff
@@ -0,0 +1,113 @@
+                throw new InvalidIndexException();
+            }
+            System.out.println("OK, the following task is removed:");
+            System.out.println(tasks.remove(index - 1));
+            System.out.println("There are " + tasks.size() + " task(s) in the list.");
+        } else if (input.equals("list")) {
+            if (tasks.size() == 0) {
+                System.out.println("You have no pending tasks on your list :)");
+            } else {
+                System.out.println("The tasks on your list are as follows:");
+                tasks.printTasks();
+            }
+        } else if (input.contains("todo")) {
+            if (input.replace("todo", "").trim().length() == 0) {
+                throw new EmptyDescriptionException("todo");
+            }
+            Task task = new Todo(input);
+            tasks.add(task);
+            System.out.println("Task added!");
+            System.out.println(task);
+            System.out.println("There are " + tasks.size() + " task(s) in the list.");
+        } else if (input.contains("deadline")) {
+            if (input.replace("deadline", "").trim().length() == 0) {
+                throw new EmptyDescriptionException("deadline");
+            } else if (trimmedText.charAt(trimmedText.length() - 1) == "/".charAt(0)) {
+                throw new MissingDateException();
+            } else if (!input.contains("/")) {
+                throw new WrongDateSyntaxException();
+            }
+            Task task = new Deadline(input);
+            if (task.getDescription().trim().length() == 0) {
+                throw new EmptyDescriptionException("deadline");
+            }
+            tasks.add(task);
+            System.out.println("Task added!");
+            System.out.println(task);
+            System.out.println("There are " + tasks.size() + " task(s) in the list.");
+        } else if (input.contains("event")) {
+            if (input.replace("event", "").trim().length() == 0) {
+                throw new EmptyDescriptionException("event");
+            } else if (trimmedText.charAt(trimmedText.length() - 1) == "/".charAt(0)) {
+                throw new MissingDateException();
+            } else if (!input.contains("/")) {
+                throw new WrongDateSyntaxException();
+            }
+            Task task = new Event(input);
+            tasks.add(task);
+            System.out.println("Task added!");
+            System.out.println(task);
+            System.out.println("There are " + tasks.size() + " task(s) in the list.");
+        } else if (input.contains("find")) {
+            if (input.replace("find", "").trim().length() == 0) {
+                throw new EmptyDescriptionException("find");
+            }
+            System.out.println("Here are some tasks that match your request:");
+            tasks.findTasks(input.substring(5));
+        } else {
+            throw (new UnidentifiedException());
+        }
+    }
+}
```

Comment:
This method could be too lengthy. Do you think you could possibly make some modifications and shorten it?

Even though it's lengthy, but I understood everything written because of how you coded and named your variables. Great job! 😄